### PR TITLE
Use pipes instead of temp-files

### PR DESCRIPTION
**Pull request by asccc**
This commit removes temp-files and uses pipes instead to pass code-contents to formatters (via stdin)
I tested all formatters and made changes to the command-lines where necessary (please add suggestions)

**Explanation:**
Using pipes reduces the IO to format a single piece of code (which can add up).

Previously formatting a single code-block required the following steps:

- create a new file
- write code-contents to the file
- pass the file to a formatter (which read from and wrote to the file, too)
- read the file-content (the formatted code) 
- delete the file

With this change, all those operations are gone.

### CODE REVIEW

File: app/CodeFormatter/GoCodeFormatter.php [hunk 1]
```diff
@@ -10,17 +10,15 @@ class GoCodeFormatter extends CodeFormatter
...
     /**
-     * @param string $file
-     *
      * @return string
      */
-    protected function getShellCommand(string $file): string
+    protected function getShellCommand(): string
     {
-        return 'gofmt -w ' . $file;
+        return 'gofmt';
```
Comment (dominikbraun) [inline app/CodeFormatter/GoCodeFormatter.php:22]: What happend to my `-w`? :D

Comment (asccc) [reply app/CodeFormatter/GoCodeFormatter.php:22]: We want the result in stdout :D

Comment (dominikbraun) [reply app/CodeFormatter/GoCodeFormatter.php:22]: Ok :)

**Review by dominikbraun — APPROVED** [review]
LGTM

**Comment by wookiefriseur** [comment]
> LGTM

Let's go to Mars?

**Review by wookiefriseur — APPROVED** [review]
Good change.

Only use case for files or DB I can think of would be preventing repeated formatting (storing it for instance as `<HASH OF INPUT>` -> `<FULL OUTPUT>`) to use a lookup instead of calling the formatter.

But since the files are only temporary and the code blocks are really small, storing and hashing would just be unnecessary overhead.

**Comment by asccc** [comment]
Caching does not depend on the processing though, we can still cache the result if we want to 😀

**Comment by asccc** [comment]
@jr-cologne what do you think? can we merge this pr?
I already made some improvements the bbcode parser, but the changes i made in my parser-branch depend on the changes i made in this pr and i probably have to rebase my branch 😆

**Review by jr-cologne — APPROVED** [review]

**Comment by jr-cologne** [comment]
Sorry for letting you wait. I didn't have the time yesterday.

This change is absolutely great. The idea to somehow remove the usage of temporary files is something which would never have come to my mind.
Thank you for bringing this project forward so much!

**Pull request merged**